pkg/proto/nbcontract/v1/customcloudconfig.proto

```diff
@@ -2,18 +2,15 @@ syntax = "proto3";
 package nbcontract.v1;
 
 message CustomCloudConfig {
-  // IsAKSCustomCloud is a flag to indicate if it is AKS custom cloud
-  bool is_aks_custom_cloud = 1;
-
-  // InitFilePath is the path to the file that contains the init script
-  string init_file_path = 2;
+  // Name of custom cloud
+  string custom_cloud_env_name = 1;
```

Comment:
is_aks_custom_cloud is quite messy per my shallow understand.
I see this in the cse_cmd
``` {{/* both CLOUD and ENVIRONMENT have special values when IsAKSCustomCloud == true */}}
{{/* CLOUD uses AzureStackCloud and seems to be used by kubelet, k8s cloud provider */}}
{{/* target environment seems to go to ARM SDK config */}}
{{/* not sure why separate/inconsistent? */}}
{{/* see GetCustomEnvironmentJSON for more weirdness. */}}
TARGET_CLOUD="{{- if IsAKSCustomCloud -}} AzureStackCloud {{- else -}} {{GetTargetEnvironment}} {{- end -}}"
TARGET_ENVIRONMENT="{{GetTargetEnvironment}}"
``` 

And in /aks-rp/protos/hcp/types/managedcluster/containerservice/v1/properties_methods.go, it also depends on p.CustomCloudEnv != nil. So seems not be able to compute from single custom_cloud_env_name?
I could be wrong though, because the logic here crosses over with others.
```// IsAKSCustomCloud checks if it's in AKS custom cloud
func (p *Properties) IsAKSCustomCloud() bool {
	return p.CustomCloudEnv != nil &&
		strings.EqualFold(p.CustomCloudEnv.Name, AKSCustomCloudName)
}
```

Reply:
I think they are just checking `p.CustomCloudEnv != nil`, because they are accessing `p.CustomCloudEnv.Name` in the next statement to avoid nil pointer dereference. There are no other checks on other fields in the config, so I think it is purely determined by the name.

Reply:
makes sense to me :)